docs/api.rst

```diff
@@ -0,0 +1,771 @@
+        "values": ["British Columbia", "Ontario"]
+    }
+
+In the filter object above, you specify `provinceOfResidence` as the field, and `values`
+to be the `["British Columbia", "Ontario"]`, `in` as the operator.
+
+This filter object asks server to find records whose `provinceOfResidence` is one of
+`British Columbia` or `Ontario`.
+
+.. note::
+    For Type II filter object, `in` is the only supported operator. Also, you specify `values`,
+    instead of `value`.
+
+--------------
+Fetch Datasets
+--------------
+Description: To fetch all of the datasets, make a query to datasets/search endpoint with an empty body. You need datasetId for subsequent queries to other endpoints.
+
+Query:
+
+.. code-block:: json
+
+   {}
+
+
+--------------
+Sample Query I
+--------------
+Description: Fetch all of the items from the specified dataset from any clinical or pipeline metadata service endpoint.
+
+Query:
+
+.. code-block:: json
+
+    {
+        "datasetId": "WyIxa2dlbm9tZSJd"
+    }
+
+---------------
+Sample Query II
+---------------
+
+Description: Fetch all of the items from the patients/search endpoint, whose gender is female.
+
+.. note::
+    We support a number of operators in the filters,
+    which include: >, <, >=, <=, =, !=, and contains.
+
+Query:
+
+.. code-block:: json
+
+    {
+        "datasetId": "WyIxa2dlbm9tZSJd",
+        "filters": [
+            {
+                "field": "gender",
+                "operator": "=",
+                "value": "female"
+            }
+        ]
+    }
+
+
+----------------
+Sample Query III
+----------------
+
+Description: Fetch all of the items from the patients/search endpoint, whose gender is female and whose ethnicity is NOT ‘GBR’.
+
+.. note::
+    You can specify more than one filter under filters.
+
+.. code-block:: json
+
+    {
+        "datasetId": "WyIxa2dlbm9tZSJd",
+        "filters": [
+            {
+                "field": "gender",
+                "operator": "=",
+                "value": "female"
+            },
+            {
+                "field": "ethnicity",
+                "operator": "!=",
+                "value": "GBR"
+            }
+        ]
+    }
+
+
+---------------
+Sample Query IV
+---------------
+
+Description: Fetch all of the items from the patients/search endpoint, whose ethnicity is one of GBR, FIN or ESN.
+
+.. note::
+    Note that this query is very different from the previous ones.
+    To specify a list of values you are interested in, you need to:
+    Specify ``values`` in your filters, instead of ``value``.
+    Specify a list of values that you are interested in.
+    Specify ``in`` as the operator.
+
+.. code-block:: json
+
+    {
+        "datasetId": "WyIxa2dlbm9tZSJd",
+        "filters": [
+            {
+                "field": "ethnicity",
+                "operator": "in",
+                "values": [
+                    "GBR",
+                    "FIN",
+                    "ESN"
+                ]
```

Comment:
We must not change anything, it is all good. However, complement test, like "not in" might be needed  in the long run.

Reply:
i will take a look at not in now

Reply:
In commit https://github.com/CanDIG/candig-server/pull/223/commits/f4068b800f669665a38981dffb8650642c31aa5d i fixed everything else except for this issue. this will be left to a future release